modules/snooker/src/main/ts/ephox/snooker/resize/Adjustments.ts

```diff
@@ -13,6 +13,20 @@ import { BarPositions, RowInfo } from './BarPositions';
 
 const sumUp = (newSize: number[]) => Arr.foldr(newSize, (b, a) => b + a, 0);
 
+const recalculateAndApply = (warehouse: Warehouse, widths: number[], tableSize: TableSize): void => {
+  // Set the width of each cell based on the column widths
```

Comment:
I am wondering if this is a possible alternative to avoid the `let` binding:
```
const recalculate = Warehouse.hasColumns(warehouse) ? Recalculations.recalculateWidthForColumns : Recalculations.recalculateWidthForCells;
const newSizes = recalculate(warehouse, widths);
```